google/cloud/storage/transfer_manager.py

```diff
@@ -567,13 +580,17 @@ def upload_many_from_filenames(
     """
     if blob_constructor_kwargs is None:
         blob_constructor_kwargs = {}
+    if additional_blob_attributes is None:
+        additional_blob_attributes = {}
 
     file_blob_pairs = []
 
     for filename in filenames:
         path = os.path.join(source_directory, filename)
         blob_name = blob_name_prefix + filename
         blob = bucket.blob(blob_name, **blob_constructor_kwargs)
+        for prop, value in additional_blob_attributes.items():
+            setattr(blob, prop, value)
```

Comment:
I'm curious how the blob properties are actually picked up in `blob._changes`. I thought we're relying on `blob._changes` to get writable metadata for the upload arguments
https://github.com/googleapis/python-storage/blob/c5a983d5a0b0632811af86fb64664b4382b05512/google/cloud/storage/blob.py#L1708-L1713

Reply:
In the _PropertyMixin, all the fields set as a _scalar_property have a custom setter that updates _changes. This solution using setattr will also activate that.

Reply:
Ah I see, that makes sense, thanks! One last nit on the naming,  maybe consider `additional_blob_properties` for consistency? Property is used in the official docs and in the lib.

Reply:
Haha, that was actually the name in my first revision.

I changed it to "attributes" because it's not only _scalar_property but any attribute that we can set with setattr. Also, since we're using setattr the nomenclature matches.